core/src/com/group66/game/screens/StartScreen.java

```diff
@@ -109,19 +109,39 @@ public void setupButtons() {
         // revert the checked state.
         setName.addListener(new ChangeListener() {
             public void changed(ChangeEvent event, Actor actor) {
-                NameInputListener listener = new NameInputListener(dynamicSettings);
+                NameInputListener listener = new NameInputListener(BustaMove.getGameInstance().getDynamicSettings());
                 Gdx.input.getTextInput(listener, "Enter your name", "", "");
             }
         });
         startButton.addListener(new ChangeListener() {
             public void changed(ChangeEvent event, Actor actor) {
-                if (!dynamicSettings.getName().equals("")) {
-                    BallAnimationLoader.profileManager.readData(dynamicSettings.getName(), dynamicSettings);
+                if (!BustaMove.getGameInstance().getDynamicSettings().getName().equals("")) {
```

Comment:
Correct way should be: !"".equals(BustaMove.getGameInstance().getDynamicSettings().getName())
Because getName could be null, causing a null pointer exception. It feels weird, but it is safer.